executor/aggregate.go

```diff
@@ -816,6 +837,104 @@ func (e *HashAggExec) getPartialResults(groupKey string) []aggfuncs.PartialResul
 	return partialResults
 }
 
+func (e *HashAggExec) initRuntimeStats() {
+	if e.runtimeStats != nil && e.stats == nil {
+		e.stats = &HashAggRuntimeStats{
+			PartialNum:        e.ctx.GetSessionVars().HashAggPartialConcurrency(),
+			FinalNum:          e.ctx.GetSessionVars().HashAggFinalConcurrency(),
+			PartialTaskNum:    0,
+			FinalTaskNum:      0,
+			PartialWallTime:   0,
+			FinalWallTime:     0,
+			PartialWorkerTime: make([]time.Duration, e.ctx.GetSessionVars().HashAggPartialConcurrency()),
+			FinalWorkerTime:   make([]time.Duration, e.ctx.GetSessionVars().HashAggFinalConcurrency()),
+		}
+		e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.RegisterStats(e.id, e.stats)
+	}
+}
+
+// HashAggRuntimeStats record the HashAggExec runtime stat
+type HashAggRuntimeStats struct {
+	PartialNum int
+	FinalNum   int
+
+	PartialTaskNum int64
+	FinalTaskNum   int64
+
+	PartialWallTime time.Duration
+	FinalWallTime   time.Duration
+
+	PartialWorkerTime []time.Duration
+	FinalWorkerTime   []time.Duration
+}
+
+func (e *HashAggRuntimeStats) String() string {
+	var result string
+	if e.PartialWallTime != 0 {
+		result += fmt.Sprintf("partial_worker:{wall_time:%s, concurrency:%d, task_num:%d", e.PartialWallTime, e.PartialNum, e.PartialTaskNum)
+	}
+	if e.PartialWorkerTime != nil {
+		sort.Slice(e.PartialWorkerTime, func(i, j int) bool { return e.PartialWorkerTime[i] < e.PartialWorkerTime[j] })
+		n := len(e.PartialWorkerTime)
+		if n != 0 {
+			result += fmt.Sprintf(", partial_max:%v, p95:%v", e.PartialWorkerTime[n-1], e.PartialWorkerTime[n*19/20])
```

Comment:
```suggestion
			result += fmt.Sprintf(", max:%v, p95:%v", e.PartialWorkerTime[n-1], e.PartialWorkerTime[n*19/20])
```

Already has `partial` prefix before, so it is redundant here?

Reply:
BTW, doesn't have total worker time?